### PR TITLE
Add Support for Cloud Role Name for Quickpulse SDK

### PR DESCRIPTION
**Pull request by rambhatt-msft**
Fix Issue # .

## Changes
Added support for Cloud Role Name for Quickpulse SDK

### Checklist
- [x] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

File: WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs [hunk 1]
```diff
@@ -1187,10 +1215,12 @@ public void QuickPulseServiceClientSubmitsInstanceNameToServiceWithPing()
         {
```
Comment (tokaplan) [inline WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs:1215]: QuickPulseServiceClientSubmitsInstanceNameToServiceWithPing now checks both instance and role, let's rename the test to reflect its new purpose.

File: WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs [hunk 2]
```diff
@@ -1216,9 +1247,11 @@ public void QuickPulseServiceClientSubmitsInstanceNameToServiceWithSubmitSamples
             // ARRANGE
```
Comment (tokaplan) [inline WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs:1247]: QuickPulseServiceClientSubmitsInstanceNameToServiceWithSubmitSamples now checks both instance and role, let's rename the test to reflect its new purpose.

File: WEB/Src/PerformanceCollector/PerformanceCollector/QuickPulseTelemetryModule.cs [hunk 3]
```diff
@@ -277,6 +277,23 @@ private static string GetInstanceName(TelemetryConfiguration configuration)
             return string.IsNullOrWhiteSpace(fakeItem.Context?.Cloud?.RoleInstance) ? Environment.MachineName : fakeItem.Context.Cloud.RoleInstance;
         }
 
+        private static string GetRoleName(TelemetryConfiguration configuration)
```
Comment (tokaplan) [inline WEB/Src/PerformanceCollector/PerformanceCollector/QuickPulseTelemetryModule.cs:280]: Let's not have a separate GetRoleName method doing exactly the same as GetInstanceName that we already have. Let's have a single method that fetches both values, call it only once, and then use its outputs to assign to both instance and roleName.

**Review by tokaplan — CHANGES_REQUESTED** [review]

File: WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs [hunk 4]
```diff
@@ -1208,17 +1238,47 @@ public void QuickPulseServiceClientSubmitsInstanceNameToServiceWithPing()
...
+                string.Empty,
+                new Clock(),
+                false,
+                0);
+
+            // ACT
+            CollectionConfigurationInfo configurationInfo;
+            serviceClient.Ping(Guid.NewGuid().ToString(), DateTimeOffset.UtcNow, string.Empty, string.Empty, out configurationInfo);
+
+            // SYNC
```
Comment (tokaplan) [inline WEB/Src/PerformanceCollector/Perf.Tests/QuickPulse/QuickPulseServiceClientTests.cs:1263]: [minor] ARRANGE - ACT - ASSERT is the constant format here, waiting should probably be under ASSERT.

**Review by tokaplan — APPROVED** [review]
Approved

**Comment by TimothyMothra** [comment]
LGTM.

Please update the Changelog under the section "VNext"

**Review by TimothyMothra — APPROVED** [review]

**Comment by rambhatt-msft** [comment]
@cijothomas 

> LGTM.
> Wondering what is the actual UI change planned with this feature. Is it to do filtering? or show rolename, along with roleinstance in the charts?

Yes. The UX will let you see the roles and filter by them too. This is especially in our use case with AKS clusters where we can restrict roles that we dont want sending data and choking our service

**Pull request merged**